src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java

```diff
@@ -451,6 +481,200 @@ protected final void verify(Checker checker,
         checker.destroy();
     }
 
+    /**
+     * Performs verification of the file with the given file path against google config.
+     * It uses the specified list of modules to load them from google config for validation.
+     * And also uses the array of expected messages for verification.
+     *
+     * @param listOfModules list of modules to load from google config.
+     * @param filePath file path to verify.
+     * @throws Exception if exception occurs during verification process.
+     */
+    public final void verifyWithGoogleConfigParser(Map<String, String[]> listOfModules,
+           String filePath) throws Exception {
+        final DefaultConfiguration googleConfig = createGoogleConfig(listOfModules);
+        final TestInputConfiguration testInputConfiguration =
+                InlineConfigParser.parseGoogleConfig(filePath);
+        final List<String> actualViolations = getActualViolationsForFile(googleConfig, filePath);
+
+        verifyViolations(filePath, testInputConfiguration.getViolations(), actualViolations);
+    }
+
+    /**
+     * Tests the file with the check config.
+     *
+     * @param config check configuration.
+     * @param file input file path.
+     * @return list of actual violations.
+     * @throws Exception if exception occurs during verification process.
+     */
+    private List<String> getActualViolationsForFile(DefaultConfiguration config,
+          String file) throws Exception {
+        stream.flush();
+        stream.reset();
+        final List<File> files = Collections.singletonList(new File(file));
+        final Checker checker = createChecker(config);
+        final Map<String, List<String>> actualViolations =
+                getActualViolations(checker.process(files));
+        checker.destroy();
+        return actualViolations.getOrDefault(file, new ArrayList<>());
+    }
+
+    /**
+     * Returns the actual violations for each file that has been
+     * checked against google config modules.
```

Comment:
please make javadoc same as source class, it is copy paste and it should be full copy paste that is easy to detect and resolve later on.

Reply:
done 

https://github.com/checkstyle/checkstyle/blob/a0f82691156c959fbb7dccd991c3e5c4f83022e1/src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java#L523-L531